### PR TITLE
Add possibility to use configured Service Account Name for init job.

### PR DESCRIPTION
**Pull request by przemyslawandruszewski**
### SUMMARY
Possibility to use Service Account Name by init-job.

### TESTING INSTRUCTIONS
`helm install superset .`

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by przemyslawandruszewski** [comment]
Hi @craig-rueda! Could you please review?

**Comment by wiktor2200** [comment]
Could you please take a look on my PR and issue? 
PR: https://github.com/apache/superset/pull/17880
issue: https://github.com/apache/superset/issues/17879
I've extended this feature (added service account creation if needed and added helpers for managing multiple conditions).
If you want, I can add your PR to mine, so custom service account will be available everywhere.

**Comment by przemyslawandruszewski** [comment]
Hey @wiktor2200, Sure I have just quickly looked at your PR so it looks nice, I will have maybe 1/2 suggestions so if you are OK, let's merge the changes? :)

**Comment by wiktor2200** [comment]
Ok, so I will add your changes to my PR. Sure feel free to add your suggestions, all feedback is welcome :)

**Comment by amitmiran137** [comment]
> Ok, so I will add your changes to my PR. Sure feel free to add your suggestions, all feedback is welcome :)

after this one: https://github.com/apache/superset/pull/17886 is merged. so rebase from master and get that fix in

File: helm/superset/templates/init-job.yaml [hunk 1]
```diff
@@ -31,6 +31,9 @@ spec:
         {{ toYaml .Values.init.podAnnotations | nindent 8 }}
       {{- end }}
     spec:
+      {{- if .Values.serviceAccountName }}
```
Comment (craig-rueda) [inline helm/superset/templates/init-job.yaml:34]: Can you add a default value, with a comment around it in the `values.yaml`?

**Comment by wiktor2200** [comment]
Thanks a lot for help and co-op!
I've missed typo in my PR, already fixed it in #17920. I'm mentioning it here to inform you on current progress. :)

**Pull request closed without merging**